--- src/Makefile.am~	2005-03-21 15:44:27.000000000 +0100
+++ src/Makefile.am	2009-02-14 00:57:27.000000000 +0100
@@ -17,8 +17,8 @@
 libscim_input_pad_la_LDFLAGS = -version-info $(SCIM_INPUT_PAD_CURRENT):$(SCIM_INPUT_PAD_REVISION):$(SCIM_INPUT_PAD_AGE) \
 			  -export-dynamic \
 			  -rpath $(libdir) \
-			  @LIBTOOL_EXPORT_OPTIONS@ \
-			  @LTLIBINTL@ \
+			  @LIBTOOL_EXPORT_OPTIONS@
+libscim_input_pad_la_LIBADD = @LTLIBINTL@ \
 			  @SCIM_LIBS@
 
 moduledir		= @SCIM_MODULEDIR@/Helper
@@ -29,8 +29,7 @@
 input_pad_la_LDFLAGS	= -avoid-version \
 		     	  -rpath $(moduledir) \
 			  -module \
-			  @LIBTOOL_EXPORT_OPTIONS@ \
-			  @LTLIBINTL@ \
-			  @GTK2_LIBS@
-
-input_pad_la_LIBADD	= libscim-input-pad.la
+			  @LIBTOOL_EXPORT_OPTIONS@
+input_pad_la_LIBADD	= @LTLIBINTL@ \
+			  @GTK2_LIBS@ @SCIM_LIBS@ \
+			  libscim-input-pad.la
--- src/Makefile.in~	2005-12-06 04:12:44.000000000 +0100
+++ src/Makefile.in	2009-02-14 00:58:28.000000000 +0100
@@ -291,8 +291,8 @@
 libscim_input_pad_la_LDFLAGS = -version-info $(SCIM_INPUT_PAD_CURRENT):$(SCIM_INPUT_PAD_REVISION):$(SCIM_INPUT_PAD_AGE) \
 			  -export-dynamic \
 			  -rpath $(libdir) \
-			  @LIBTOOL_EXPORT_OPTIONS@ \
-			  @LTLIBINTL@ \
+			  @LIBTOOL_EXPORT_OPTIONS@
+libscim_input_pad_la_LIBADD = @LTLIBINTL@ \
 			  @SCIM_LIBS@
 
 moduledir = @SCIM_MODULEDIR@/Helper
@@ -302,11 +302,10 @@
 input_pad_la_LDFLAGS = -avoid-version \
 		     	  -rpath $(moduledir) \
 			  -module \
-			  @LIBTOOL_EXPORT_OPTIONS@ \
-			  @LTLIBINTL@ \
-			  @GTK2_LIBS@
-
-input_pad_la_LIBADD = libscim-input-pad.la
+			  @LIBTOOL_EXPORT_OPTIONS@
+input_pad_la_LIBADD = @LTLIBINTL@ \
+			  @GTK2_LIBS@ @SCIM_LIBS@ \
+			  libscim-input-pad.la
 all: all-am
 
 .SUFFIXES:
